source/org/zfin/expression/repository/HibernateExpressionRepository.java

```diff
@@ -1462,437 +1382,434 @@ public void deleteExpressionResultPerFigure(ExpressionResult2 result, Figure fig
+		Set<ExpressionStructure> distinctStructures = new HashSet<>();
+		for (ExpressionResult2 expressionResult : expressionResults) {
+			Set<ExpressionStructure> expressionStructureSet = instantiateExpressionStructure(expressionResult, publication);
+			for (ExpressionStructure structure : expressionStructureSet) {
+				// only create structures that are not already on the pile.
+				if (distinctStructures.add(structure))
+					createPileStructure(structure);
+			}
+		}
+
+	}
+
+	private Set<ExpressionStructure> instantiateExpressionStructure(ExpressionResult2 expressionResult, Publication publication) {
+		Set<ExpressionStructure> structureList = new HashSet<>();
+		if (expressionResult.isEap()) {
+			for (ExpressionPhenotypeTerm quality : expressionResult.getPhenotypeTermSet()) {
+				ExpressionStructure structure = getExpressionStructure(expressionResult, publication);
+				structure.setEapQualityTerm(quality.getQualityTerm());
+				structure.setTag(quality.getTag());
+				structureList.add(structure);
+			}
+		} else {
+			structureList.add(getExpressionStructure(expressionResult, publication));
+		}
+		return structureList;
+	}
+
+	private ExpressionStructure getExpressionStructure(ExpressionResult2 expressionResult, Publication publication) {
+		ExpressionStructure structure = new ExpressionStructure();
+		structure.setDate(new Date());
+		structure.setPerson(ProfileService.getCurrentSecurityUser());
+		structure.setPublication(publication);
+		structure.setSuperterm(expressionResult.getSuperTerm());
+		structure.setExpressionFound(expressionResult.isExpressionFound());
+		GenericTerm subTerm = expressionResult.getSubTerm();
+		if (subTerm != null) {
+			structure.setSubterm(subTerm);
+		}
+		return structure;
+	}
+
+	private List<ExpressionResult2> getAllExpressionResults(String publicationID) {
+		Session session = HibernateUtil.currentSession();
+		String hql = "select distinct expression from ExpressionResult2 expression where" +
+			"          expression.expressionFigureStage.expressionExperiment.publication.id= :publicationID";
+		Query<ExpressionResult2> query = session.createQuery(hql, ExpressionResult2.class);
+		query.setParameter("publicationID", publicationID);
+		return query.list();
+
+	}
+
+	/**
+	 * Create a new structure - post-composed - for the structure pile.
+	 *
+	 * @param structure structure
+	 */
+	public void createPileStructure(ExpressionStructure structure) {
+		Session session = HibernateUtil.currentSession();
+		session.save(structure);
+	}
+
+	/**
+	 * Retrieve Expressions for a given term.
+	 *
+	 * @param term term
+	 * @return list of expressions
+	 */
+	@Override
+	public List<ExpressionResult> getExpressionsWithEntity(GenericTerm term) {
+		String hql = "select distinct expression from ExpressionResult expression where " +
+			"(entity.superterm = :term OR entity.subterm = :term) " +
+			" AND expressionFound = :expressionFound ";

```

Comment:
It's personal preference, but I like to embed the "true" in the hql myself, so it reads "... AND expressionFound = true ..."

There are a few other places in the refactor that contain query.setParameter("someTerm", true) too.

Reply:
I agree! If a parameter is fixed / hard-coded then do it right in the hql. Only use the setParameter() method if it gets passed in and can have multiple values.